dapps/assets/src/components/digitaltwins/DigitalTwins.ts

```diff
@@ -99,23 +130,37 @@ export default class DigitalTwinsComponent extends mixins(EvanComponent) {
       const offset = 200;
       let bottomOfWindow = clientHeight + scrollTop >= scrollHeight - offset;
 
-      if (bottomOfWindow && typeof this.fetchMore === 'function') {
-        this.fetchMore();
+      if (bottomOfWindow) {
+        this.performFetchMore();
       }
     },
     100,
     { trailing: true }
   );
 
+  /**
+   * Helper method to keep class this-context for debounced search.
+   */
+  performSearch() {
+    if (typeof this.search === 'function') {
+      this.search(this.searchTerm, { sortBy: this.sortBy, reverse: this.reverse, type: this.selectedFilter });
+    }
+  }
+
+  /**
+   * Helper method to keep class this-context for debounced fetch.
+   */
+  performFetchMore() {
+    if (typeof this.fetchMore === 'function') {
+      this.fetchMore({ sortBy: this.sortBy, reverse: this.reverse, type: this.selectedFilter });
+    }
+  }
+
   /**
    * Debounce the search for 0.25s.
```

Comment:
Imo avoid such specific values in comments. Will be hard to maintain.
```suggestion
   * Debounce the search.
```